File: SDKs/googleapis.com/sasportal/v1alpha1/python/src/sdk/models/shared/sasportalinstallationparams.py
from dataclasses import dataclass, field
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils

class SasPortalInstallationParamsHeightTypeEnum(str, Enum):
    HEIGHT_TYPE_UNSPECIFIED = "HEIGHT_TYPE_UNSPECIFIED"
    HEIGHT_TYPE_AGL = "HEIGHT_TYPE_AGL"
    HEIGHT_TYPE_AMSL = "HEIGHT_TYPE_AMSL"


@dataclass_json
@dataclass
class SasPortalInstallationParams:
    r"""SasPortalInstallationParams
    Information about the device installation parameters.
    """
    
    antenna_azimuth: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('antennaAzimuth') }})
    antenna_beamwidth: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('antennaBeamwidth') }})
    antenna_downtilt: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('antennaDowntilt') }})
    antenna_gain: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('antennaGain') }})
    antenna_model: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('antennaModel') }})
    cpe_cbsd_indication: Optional[bool] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('cpeCbsdIndication') }})
    eirp_capability: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('eirpCapability') }})
    height: Optional[float] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('height') }})
    height_type: Optional[SasPortalInstallationParamsHeightTypeEnum] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('heightType') }})
    horizontal_accuracy: Optional[float] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('horizontalAccuracy') }})
    indoor_deployment: Optional[bool] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('indoorDeployment') }})
    latitude: Optional[float] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('latitude') }})
    longitude: Optional[float] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('longitude') }})
    vertical_accuracy: Optional[float] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('verticalAccuracy') }})
    

File: SDKs/googleapis.com/sasportal/v1alpha1/typescript/src/sdk/models/shared/sasportalinstallationparams.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum SasPortalInstallationParamsHeightTypeEnum {
    HeightTypeUnspecified = "HEIGHT_TYPE_UNSPECIFIED",
    HeightTypeAgl = "HEIGHT_TYPE_AGL",
    HeightTypeAmsl = "HEIGHT_TYPE_AMSL"
}


// SasPortalInstallationParams
/** 
 * Information about the device installation parameters.
**/
export class SasPortalInstallationParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=antennaAzimuth" })
  antennaAzimuth?: number;

  @SpeakeasyMetadata({ data: "json, name=antennaBeamwidth" })
  antennaBeamwidth?: number;

  @SpeakeasyMetadata({ data: "json, name=antennaDowntilt" })
  antennaDowntilt?: number;

  @SpeakeasyMetadata({ data: "json, name=antennaGain" })
  antennaGain?: number;

  @SpeakeasyMetadata({ data: "json, name=antennaModel" })
  antennaModel?: string;

  @SpeakeasyMetadata({ data: "json, name=cpeCbsdIndication" })
  cpeCbsdIndication?: boolean;

  @SpeakeasyMetadata({ data: "json, name=eirpCapability" })
  eirpCapability?: number;

  @SpeakeasyMetadata({ data: "json, name=height" })
  height?: number;

  @SpeakeasyMetadata({ data: "json, name=heightType" })
  heightType?: SasPortalInstallationParamsHeightTypeEnum;

  @SpeakeasyMetadata({ data: "json, name=horizontalAccuracy" })
  horizontalAccuracy?: number;

  @SpeakeasyMetadata({ data: "json, name=indoorDeployment" })
  indoorDeployment?: boolean;

  @SpeakeasyMetadata({ data: "json, name=latitude" })
  latitude?: number;

  @SpeakeasyMetadata({ data: "json, name=longitude" })
  longitude?: number;

  @SpeakeasyMetadata({ data: "json, name=verticalAccuracy" })
  verticalAccuracy?: number;
}
